### PR TITLE
Super Staff Bros: Implement bugfixes and balance updates

### PR DESCRIPTION
**Pull request by HisuianZoroark**
TODO: A lot

### CODE REVIEW

File: data/mods/gen9ssb/moves.ts [hunk 1]
```diff
@@ -2103,6 +2103,9 @@ export const Moves: import('../../../sim/dex-moves').ModdedMoveDataTable = {
 			this.add('-anim', source, 'Dizzy Punch', target);
 			this.add('-anim', source, 'Bulk Up', source);
 		},
+		onHit(pokemon, source) {
+			source.addVolatile('cringedadjoke');
+		},
```
Comment (urkerab) [inline data/mods/gen9ssb/moves.ts:2108]:
```suggestion
		self: {
			volatileStatus: 'cringedadjoke';
		},
```

File: data/mods/gen9ssb/moves.ts [hunk 2]
```diff
@@ -3610,7 +3585,9 @@ export const Moves: import('../../../sim/dex-moves').ModdedMoveDataTable = {
 				this.boost({[boost]: 1}, pokemon);
 			}
 			this.add(`c:|${getName((pokemon.illusion || pokemon).name)}|Ope! Wrong button, sorry.`);
-			const unloweredStat = this.sample(Object.keys(pokemon.boosts).filter(x => x !== ('evasion' as BoostID)));
+			const unloweredStat = this.sample(
+				Object.keys(pokemon.boosts).filter(x => x !== ('evasion' as BoostID || 'accuracy' as BoostID))
```
Comment (KrisXV) [inline data/mods/gen9ssb/moves.ts:3589]:
```suggestion
				Object.keys(pokemon.boosts).filter(x => !['evasion', 'accuracy'].includes(x as BoostID))
```

File: data/mods/gen9ssb/random-teams.ts [hunk 3]
```diff
@@ -1106,7 +1100,7 @@ export class RandomStaffBrosTeams extends RandomTeams {
 		this.enforceNoDirectCustomBanlistChanges();
 
 		const team: PokemonSet[] = [];
-		const debug: string[] = []; // Set this to a list of SSB sets to override the normal pool for debugging.
+		const debug: string[] = ['Lily']; // Set this to a list of SSB sets to override the normal pool for debugging.
```
Comment (KrisXV) [inline data/mods/gen9ssb/random-teams.ts:1103]: wut

Comment (HisuianZoroark) [reply data/mods/gen9ssb/random-teams.ts:1103]: oops

File: data/mods/gen9ssb/moves.ts (hunk 1 above)
Comment (urkerab) [reply data/mods/gen9ssb/moves.ts]: Oops, sorry for the typo!

File: server/chat-plugins/randombattles/ssb.ts [hunk 4]
```diff
@@ -183,8 +183,13 @@ function generateSSBItemInfo(set: SSBSet, dex: ModdedDex, baseDex: ModdedDex) {
 
 function generateSSBAbilityInfo(set: SSBSet, dex: ModdedDex, baseDex: ModdedDex) {
 	let buf = ``;
-	if (!Array.isArray(set.ability) && !baseDex.abilities.get(set.ability).exists) {
-		const sigAbil = Dex.deepClone(dex.abilities.get(set.ability));
+	const customMegaAbilities = ['Sableye', 'Ampharos'];
+	if (!Array.isArray(set.ability) && (customMegaAbilities.includes(set.species) || !baseDex.abilities.get(set.ability).exists)) {
```
Comment (KrisXV) [inline server/chat-plugins/randombattles/ssb.ts:187]:
```suggestion
	if (!Array.isArray(set.ability) &&
		(customMegaAbilities.includes(set.species) || !baseDex.abilities.get(set.ability).exists)) {
```

File: server/chat-plugins/randombattles/ssb.ts [hunk 5]
```diff
@@ -183,8 +183,13 @@ function generateSSBItemInfo(set: SSBSet, dex: ModdedDex, baseDex: ModdedDex) {
...
+		let sigAbil = Dex.deepClone(dex.abilities.get(set.ability));
+		if (customMegaAbilities.includes(set.species)) {
+			let megaAbil = dex.species.get(`${set.species}-Mega`).abilities[0];
```
Comment (KrisXV) [inline server/chat-plugins/randombattles/ssb.ts:190]:
```suggestion
			const megaAbil = dex.species.get(`${set.species}-Mega`).abilities[0];
```

**Comment by HisuianZoroark** [comment]
Alright I am finally done. This is ready to be merged whenever.

**Pull request merged**